### PR TITLE
#21 Create android.yml

### PR DESCRIPTION
**Pull request by PavelStalone**

### CODE REVIEW

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: Android CI For GestureTranslator
+
+on:
+  push:
+    branches: 
+    - main
+    - release/*
+    - feature/*
```
Comment (jacksever) [inline .github/workflows/android.yml:8]:
```suggestion
    - feature/*
    - bugfix/*
    - hotfix/*
```

File: .github/workflows/android.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  pull_request:
+    branches: [ "main" ]
```
Comment (jacksever) [inline .github/workflows/android.yml:10]: Repeat what was done above

File: .github/workflows/android.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    
+    - name: set up JDK 17
```
Comment (jacksever) [inline .github/workflows/android.yml:19]:
```suggestion
    - name: Set up JDK 17
```

File: .github/workflows/android.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      uses: actions/cache@v2
+      with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+          restore-keys: ${{ runner.os }}-gradle-
+          
+    - name: Gradle check
+      run: ./gradlew check
+      
+    - name: Build with Gradle
```
Comment (jacksever) [inline .github/workflows/android.yml:39]: It is better to put it into parallel job :)

File: .github/workflows/android.yml (hunk 2 above)
Comment (PoroshinGA) [inline .github/workflows/android.yml:10]: Why there's only one branch

**Review by jacksever — CHANGES_REQUESTED** [review]
Please check my comments

**Review by PoroshinGA — CHANGES_REQUESTED** [review]

File: .github/workflows/android.yml (hunk 4 above)
Comment (PavelStalone) [reply .github/workflows/android.yml:39]: Wait... What?

Comment (jacksever) [reply .github/workflows/android.yml:39]: Repeat all code blocks that written above and create the new parallel job

File: .github/workflows/android.yml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: gradle
+
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+
```
Comment (jacksever) [inline .github/workflows/android.yml:61]:
```
      - name: Cache Gradle dependencies
        uses: actions/cache@v2
        with:
          path: ~/.gradle/caches
          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
          restore-keys: ${{ runner.os }}-gradle-
```

**Comment by PavelStalone** [comment]
PR is here: #21

**Review by jacksever — APPROVED** [review]

**Review by PoroshinGA — APPROVED** [review]

**Pull request merged**